src/glue/auth_checker.cpp

```diff
@@ -114,6 +132,26 @@ bool AuthChecker::IsUserAuthorized(const memgraph::auth::User &user,
   });
 }
 
+bool AuthChecker::IsUserAdmin(const memgraph::auth::User &user) {
+  const auto user_permissions = user.GetPermissions();
+  const std::vector<memgraph::query::AuthQuery::Privilege> privileges{
+      memgraph::query::AuthQuery::Privilege::CREATE,      memgraph::query::AuthQuery::Privilege::DELETE,
+      memgraph::query::AuthQuery::Privilege::MATCH,       memgraph::query::AuthQuery::Privilege::MERGE,
+      memgraph::query::AuthQuery::Privilege::SET,         memgraph::query::AuthQuery::Privilege::REMOVE,
+      memgraph::query::AuthQuery::Privilege::INDEX,       memgraph::query::AuthQuery::Privilege::STATS,
+      memgraph::query::AuthQuery::Privilege::AUTH,        memgraph::query::AuthQuery::Privilege::CONSTRAINT,
+      memgraph::query::AuthQuery::Privilege::DUMP,        memgraph::query::AuthQuery::Privilege::REPLICATION,
+      memgraph::query::AuthQuery::Privilege::DURABILITY,  memgraph::query::AuthQuery::Privilege::READ_FILE,
+      memgraph::query::AuthQuery::Privilege::FREE_MEMORY, memgraph::query::AuthQuery::Privilege::TRIGGER,
+      memgraph::query::AuthQuery::Privilege::CONFIG,      memgraph::query::AuthQuery::Privilege::STREAM,
+      memgraph::query::AuthQuery::Privilege::MODULE_READ, memgraph::query::AuthQuery::Privilege::MODULE_WRITE,
+      memgraph::query::AuthQuery::Privilege::WEBSOCKET};
+  return std::all_of(privileges.begin(), privileges.end(), [&user_permissions](const auto privilege) {
+    return user_permissions.Has(memgraph::glue::PrivilegeToPermission(privilege)) ==
+           memgraph::auth::PermissionLevel::GRANT;
```

Comment:
What if Has returns Neutral, isn't that also okay?

Reply:
I am not sure when it returns Neutral to be honest

Reply:
I think the logic is that permission level must be set to grant otherwise it means that no other user has actually given specific permissions to this admin user.